pkg/checks/checks.go

```diff
@@ -188,6 +260,71 @@ out:
 		}
 	}
 
+	nodeRoleToName := map[string][]string{}
+	nodeRoleToName["all"] = []string{}
+	// get a list of nodes to check for static pods
+	nodes, err := coreclient.Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes.Items {
+		nodeLabels := node.ObjectMeta.GetLabels()
+		for key := range nodeLabels {
+			if strings.HasPrefix(key, "node-role.kubernetes.io") {
+				// split "node-role.kubernetes.io/compute"
+				// should return compute
+				roleKey := strings.Split(key, "/")[1]
+				if _, ok := nodeRoleToName[roleKey]; ok {
+					nodeRoleToName[roleKey] = append(nodeRoleToName[roleKey], node.Name)
+				} else {
+					nodeRoleToName[roleKey] = []string{node.Name}
+				}
+				break
+			}
+		}
+		// put each node into all group
+		nodeRoleToName["all"] = append(nodeRoleToName["all"], node.Name)
+	}
+
+	// static pods
+	for idx, app := range applicationWhitelist["static"] {
+		log.Info(fmt.Sprintf("checking static[%d]: %s %q", idx, app.Namespace, app.Name))
+	staticlist:
+		for {
```

Comment:
Not real pleased with this logic but it does work.  It is difficult to loop forever and check that each static pod has a corresponding host.